packages/ember-routing/lib/system/router.js

```diff
@@ -636,7 +636,17 @@ var EmberRouter = EmberObject.extend(Evented, {
     // merge in any queryParams from the active transition which could include
     // queryparams from the url on initial load.
     if (this.router.activeTransition) {
-      assign(queryParams, this.router.activeTransition.queryParams);
+      var unchangedQPs = {};
```

Comment:
Can this bit of code live in a separate method (`_processActiveTransitionQueryParams` or something)?